src/Type/IntersectionType.php

```diff
@@ -34,6 +35,8 @@
 class IntersectionType implements CompoundType
 {
 
+	use NonRemoveableTypeTrait;
```

Comment:
As I see this trait being used by a lot of traits, I'm pretty sure we could actually make them support type-removing in some scenarios :) But we can leave it for later.